### PR TITLE
Script to clean instant data types

### PR DESCRIPTION
**Pull request by Ilan-Z**

### CODE REVIEW

File: scripts/custom/clean_instant.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    # Handle RFC 1123 format
+    try:
+        date = datetime.datetime.strptime(raw_input, "%a, %d %b %Y %H:%M:%S GMT")
+    except ValueError:
+        pass
+
+    if date is None:
+        return raw_input
+
+    return date.strptime("%Y-%m-%dT%H:%M:%SZ")
```
Comment (elsiehoffet-94) [inline scripts/custom/clean_instant.py:41]: I think you meant to put `date.strftime` here not `date.strptime`

File: scripts/custom/clean_dateTime.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        date = datetime.datetime.strptime(raw_input, "%Y-%m-%dT%H:%M:%S")
+        result = date.strftime("%Y%m%dT%H:%M:%S")
+    except ValueError:
+        pass
+
+    # Handle RFC 1123 format
+    try:
+        date = datetime.datetime.strptime(raw_input, "%a, %d %b %Y %H:%M:%S GMT")
+        result = date.strftime("%Y%m%dT%H:%M:%S GMT")
+
```
Comment (elsiehoffet-94) [inline scripts/custom/clean_dateTime.py:46]: Can you add a case to handle this date format "YYYY-MM-DD H:M:S"  in both scripts? You can look at the script `clean_date.py` for inspiration

File: scripts/custom/clean_dateTime.py [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    try:
+        date = datetime.datetime.strptime(raw_input, "%Y-%m-%dT%H:%M:%S")
+        result = date.strftime("%Y%m%dT%H:%M:%S")
+    except ValueError:
+        pass
+
+    # Handle YYYY-MM-DD H:M:S
+    try:
+        date = datetime.datetime.strptime(raw_input, "%Y-%m-%d %H:%M:%S")
+        result = date.strftime("%Y-%m-%d %H:%M:%S")
```
Comment (elsiehoffet-94) [inline scripts/custom/clean_dateTime.py:45]: This is not the right format to return, you need the T between date and time see : https://www.hl7.org/fhir/datatypes.html#dateTime

File: scripts/custom/clean_dateTime.py [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+        date = datetime.datetime.strptime(raw_input, "%Y-%m-%d")
+        result = date.strftime("%Y%m%d")
+    except ValueError:
+        pass
+
+    # Handle YYYY-MM-DDTH:M:S
+    try:
+        date = datetime.datetime.strptime(raw_input, "%Y-%m-%dT%H:%M:%S")
+        result = date.strftime("%Y%m%dT%H:%M:%S")
+    except ValueError:
```
Comment (elsiehoffet-94) [inline scripts/custom/clean_dateTime.py:39]: Here, this format is already fhir friendly; so you don't really need to change it. However, you need to transform something from this format `YYYY-MM-DD H:M:S` to this one `YYYY-MM-DDTH:M:S`.

File: test/custom/test_clean_dateTime.py [hunk 5]
```diff
@@ -0,0 +1,41 @@
+from scripts.custom.clean_dateTime import clean_dateTime
+
+
+def test_clean_dateTime():
+    raw_input_1 = "2015"
+    output_1 = clean_dateTime(raw_input_1)
+    print(output_1)
```
Comment (elsiehoffet-94) [inline test/custom/test_clean_dateTime.py:7]: remove all the prints

File: test/custom/test_clean_dateTime.py [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+    print(output_6)
+    assert output_6 == "2015-02-07T13:28:17+05:00"
+
+    raw_input_7 = "2015-02-07T13:28:17-05:00"
+    output_7 = clean_dateTime(raw_input_7)
+    print(output_7)
+    assert output_7 == "2015-02-07T13:28:17-05:00"
+
+
+test_clean_dateTime()
```
Comment (elsiehoffet-94) [inline test/custom/test_clean_dateTime.py:41]: remove this line

File: scripts/custom/clean_instant.py [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+        return ""
+
+    date = None
+
+    # Handle YYYYMMDD
+    try:
+        date = datetime.datetime.strptime(raw_input, "%Y%m%d")
+    except ValueError:
+        pass
+
```
Comment (elsiehoffet-94) [inline scripts/custom/clean_instant.py:19]: Can you add a case for YYYY-MM-DD?

File: scripts/custom/clean_dateTime.py [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+        pass
+
+    # Handle YYYY-MM-DD
+    try:
+        date = datetime.datetime.strptime(raw_input, "%Y-%m-%d")
+        result = date.strftime("%Y-%m-%d")
+    except ValueError:
+        pass
+
```
Comment (elsiehoffet-94) [inline scripts/custom/clean_dateTime.py:34]: Can you add the case for YYYYMMDD and YYYYMM also?

File: test/custom/test_clean_dateTime.py (hunk 6 above)
Comment (elsiehoffet-94) [reply test/custom/test_clean_dateTime.py:41]: Can you add tests for your clean_instant function?

File: scripts/custom/clean_dateTime.py [hunk 9]
```diff
@@ -0,0 +1,74 @@
...
+    try:
+        date = datetime.datetime.strptime(raw_input, "%Y-%m-%dT%H:%M:%S-%z")
+        result = date.strftime("%Y-%m-%dT%H:%M:%S%z")
+    except ValueError:
+        pass
+
+    # Handle RFC 1123 format
+    try:
+        date = datetime.datetime.strptime(raw_input, "%a, %d %b %Y %H:%M:%S GMT")
+        result = date.strftime("%Y%m%dT%H:%M:%S%z")
```
Comment (elsiehoffet-94) [inline scripts/custom/clean_dateTime.py:66]: I think the right format here is `"%Y-%m-%dT%H:%M:%S%z"`. Also, I think you should add tests for type of date.

Comment (Ilan-Z) [reply scripts/custom/clean_dateTime.py:66]: Done

File: scripts/custom/clean_dateTime.py (hunk 8 above)
Comment (Ilan-Z) [reply scripts/custom/clean_dateTime.py:34]: Done

File: scripts/custom/clean_instant.py (hunk 7 above)
Comment (Ilan-Z) [reply scripts/custom/clean_instant.py]: We must talk about it

File: test/custom/test_clean_dateTime.py (hunk 5 above)
Comment (Ilan-Z) [reply test/custom/test_clean_dateTime.py]: Done

File: test/custom/test_clean_dateTime.py (hunk 6 above)
Comment (Ilan-Z) [reply test/custom/test_clean_dateTime.py]: Done

File: test/custom/test_clean_instant [hunk 10]
```diff
@@ -0,0 +1,24 @@
...
+    assert output_2 == "2015-02-07T13:28:17+00:00"
+
+    ### RFC 1123 format test
+    raw_input_3 = "Wed, 13 Mar 2075 00:00:00 GMT"
+    output_3 = clean_instant(raw_input_3)
+    print(output_3)
+    assert output_3 == "2075-03-13T00:00:00+00:00"
+
+
+print(test_clean_instant)
```
Comment (elsiehoffet-94) [inline test/custom/test_clean_instant:24]: Remove this file

File: test/custom/test_clean_instant.py [hunk 11]
```diff
@@ -0,0 +1,18 @@
...
+    raw_input_1 = "2015-02-07 13:28:17"
+    output_1 = clean_instant(raw_input_1)
+    assert output_1 == "2015-02-07T13:28:17+02:00"
+
+    # YYYY-MM-DDTH:M:S format test
+    raw_input_2 = "2015-02-07T13:28:17"
+    output_2 = clean_instant(raw_input_2)
+    assert output_2 == "2015-02-07T13:28:17+02:00"
+
+    ### RFC 1123 format test
```
Comment (elsiehoffet-94) [inline test/custom/test_clean_instant.py:15]: Put only one # here

File: test/custom/test_clean_dateTime.py [hunk 12]
```diff
@@ -0,0 +1,53 @@
...
+    raw_input_8 = "2015-02-07T13:28:17+05:00"
+    output_8 = clean_dateTime(raw_input_8)
+    assert output_8 == "2015-02-07T13:28:17+05:00"
+
+    # YYYY-MM-DDTH:M:S-zz:zz format test
+    raw_input_9 = "2015-02-07T13:28:17-05:00"
+    output_9 = clean_dateTime(raw_input_9)
+    assert output_9 == "2015-02-07T13:28:17-05:00"
+
+    ### RFC 1123 format test
```
Comment (elsiehoffet-94) [inline test/custom/test_clean_dateTime.py:50]: Only one # here

File: test/custom/test_clean_date.py [hunk 13]
```diff
@@ -2,6 +2,10 @@
 
 
 def test_clean_date():
-    row_input = "20191103"
+    row_input = "2015-02-07T13:28:17"
     output = clean_date(row_input)
-    assert output == "2019-11-03"
+    print(output)
```
Comment (elsiehoffet-94) [inline test/custom/test_clean_date.py:7]: remove this print

File: test/custom/test_clean_date.py [hunk 14]
```diff
@@ -2,6 +2,10 @@
...
+    assert output == "2015-02-07"
+
+
+test_clean_date()
```
Comment (elsiehoffet-94) [inline test/custom/test_clean_date.py:11]: remove this line

File: scripts/custom/clean_instant.py [hunk 15]
```diff
@@ -0,0 +1,35 @@
...
+    # Handle RFC 1123 format
+    try:
+        date = datetime.datetime.strptime(raw_input, "%a, %d %b %Y %H:%M:%S GMT")
+    except ValueError:
+        pass
+
+    if date is None:
+        return raw_input
+
+    return date.strftime("%Y-%m-%dT%H:%M:%S+02:00")
```
Comment (elsiehoffet-94) [inline scripts/custom/clean_instant.py:35]: Decide whether the default is +01:00 or +02:00 but not both

File: test/custom/test_clean_date.py (hunk 13 above)
Comment (Ilan-Z) [reply test/custom/test_clean_date.py:7]: Done

File: test/custom/test_clean_date.py (hunk 14 above)
Comment (Ilan-Z) [reply test/custom/test_clean_date.py:11]: Done

File: test/custom/test_clean_dateTime.py (hunk 12 above)
Comment (Ilan-Z) [reply test/custom/test_clean_dateTime.py:50]: Done

File: test/custom/test_clean_instant (hunk 10 above)
Comment (Ilan-Z) [reply test/custom/test_clean_instant:24]: Done

File: test/custom/test_clean_instant.py (hunk 11 above)
Comment (Ilan-Z) [reply test/custom/test_clean_instant.py:15]: Done

File: scripts/custom/clean_instant.py [hunk 16]
```diff
@@ -0,0 +1,88 @@
...
+        raw_input = str(raw_input)
+    if utils.is_empty(raw_input):
+        return ""
+
+    date = None
+
+    # Handle YYYY
+    try:
+        date = datetime.datetime.strptime(raw_input, "%Y")
+        result = date.strftime("%Y-%m-%dT%H:%M:%S+02:00")
```
Comment (elsiehoffet-94) [inline scripts/custom/clean_instant.py:16]: +02:00 or +01:00?

**Pull request merged**